source/Octopus.Tentacle.Tests.Integration/Support/RetryLogMessageAssertions.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Text.RegularExpressions;
+using FluentAssertions;
+
+namespace Octopus.Tentacle.Tests.Integration.Support
+{
+    internal static class RetryLogMessageAssertions
+    {
+        static readonly Regex retryingMessageRegex = new ("An error occurred communicating with Tentacle. This action will be retried after \\d* seconds. Retry attempt \\d*. Retries will be performed for up to \\d* seconds.\\s*$", RegexOptions.Compiled | RegexOptions.Multiline);
+        static readonly Regex timeoutAfterRetriesMessageRegex = new ("Could not communicate with Tentacle after \\d* seconds. No more retries will be attempted.\\s*$", RegexOptions.Compiled | RegexOptions.Multiline);
+        static readonly Regex timeoutAfterNoRetriesMessageRegex = new ("Could not communicate with Tentacle after \\d* seconds.\\s*$", RegexOptions.Compiled | RegexOptions.Multiline);
+
+        public static void AssertRetryAttemptsLoggedButNoRetryFailureLogged(InMemoryLog logs)
+        {
+            AssertRetriedLogMessages(logs);
+            
+            // Negative assertions
+            AssertNoTimeoutAfterRetriedLogMessages(logs);
+            AssertNoTimeoutAfterNoRetriedLogMessages(logs);
+        }
+
+        public static  void AssertRetryAttemptsLoggedAndRetryFailureLogged(InMemoryLog logs)
+        {
+            AssertRetriedLogMessages(logs);
+            AssertTimeoutAfterRetriedLogMessages(logs);
+
+            // Negative assertions
+            AssertNoTimeoutAfterNoRetriedLogMessages(logs);
+        }
+
+        public static  void AssertNoRetryAttemptsLoggedAndRetryFailureLogged(InMemoryLog logs)
```

Comment:
Just a thought, but we could make this an extension method, and even make the method names match the fluent "Should" style.

Then this:
```
RetryLogMessageAssertions.AssertRetryAttemptsLoggedAndRetryFailureLogged(inMemoryLog);
```

Could be something like this:
```
inMemoryLog.ShouldHaveLoggedRetryAttemptsAndRetryFailure();
```

Reply:
Yeh nice, I'll rework this